### PR TITLE
 [CLEAN] Replace com.google.common.base.Optional with java.util.Optional

### PR DESCRIPTION
**Pull request by mino181295**
### Motivation

`com.google.common.base.Optional` and `java.util.Optional` exist in the code base at the same time.

### Changes
Replace `com.google.common.base.Optional` with `java.util.Optional`.

### CODE REVIEW

**Comment by eolivelli** [comment]
@mino181295 awesome work.

One question:
did you check if you are changing public API ?
org.apache.bookkeeper.client
org.apache.bookkeeper.client.api
and DLog API

File: site/docs/latest/api/distributedlog-api.md [hunk 1]
```diff
@@ -222,7 +222,7 @@ logConf.setRetentionPeriodHours(12);
 DistributedLogManager logManager = namespace.openLog(
         "test-log",
         Optional.of(logConf),
-        Optional.absent()
+        Optional.empty()
```
Comment (eolivelli) [inline site/docs/latest/api/distributedlog-api.md:225]: this is documentation but it looks like an API change

File: stream/distributedlog/core/src/main/java/org/apache/distributedlog/BookKeeperClientBuilder.java [hunk 2]
```diff
@@ -59,7 +59,7 @@ public static BookKeeperClientBuilder newBuilder() {
     // request timer
     private HashedWheelTimer requestTimer = null;
     // feature provider
-    private Optional<FeatureProvider> featureProvider = Optional.absent();
+    private Optional<FeatureProvider> featureProvider = Optional.empty();
```
Comment (eolivelli) [inline stream/distributedlog/core/src/main/java/org/apache/distributedlog/BookKeeperClientBuilder.java:62]: this change looks like an API change

**Comment by eolivelli** [comment]
@diegosalvi @sijie PTAL

**Comment by mino181295** [comment]
To answer your questions:
- The guava Optional was marked as @deprecated since 0.5 in favor of 
`
loadStreamConf(java.util.Optional<DistributedLogConfiguration> streamConfiguration)
`
- The `Namespace` class is marked as `@Evolving` and I think this change it's not a big deal to fix

cc @sijie

**Pull request merged**